app/models/conversation.rb

```diff
@@ -86,7 +86,7 @@ def subscribers(user)
   end
 
   def receive(user, person)
-    cnv = Conversation.find_or_create_by!(self.attributes)
+    cnv = Conversation.create_with(self.attributes).find_or_create_by!(self.attributes[:guid])
```

Comment:
`Conversation.create_with(attributes).find_or_create_by!(guid: guid)` generates: `SELECT  "conversations".* FROM "conversations"  WHERE "conversations"."guid" = 'cf7cd2e09a6001329fdd14feb5c40cd3' LIMIT 1` and doesn't fail the test for me.